hudi-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadata.java

```diff
@@ -255,38 +255,35 @@ public List<Pair<String, Option<HoodieRecord<HoodieMetadataPayload>>>> getRecord
     return result;
   }
 
-  private Map<String, Option<HoodieRecord<HoodieMetadataPayload>>> readLogRecords(HoodieMetadataMergedLogRecordReader logRecordScanner,
+  private Map<String, Option<HoodieRecord<HoodieMetadataPayload>>> readLogRecords(HoodieMetadataLogRecordReader logRecordReader,
                                                                                   List<String> keys,
                                                                                   boolean fullKey,
                                                                                   List<Long> timings) {
     HoodieTimer timer = HoodieTimer.start();
 
-    if (logRecordScanner == null) {
+    if (logRecordReader == null) {
       timings.add(timer.endTimer());
       return Collections.emptyMap();
     }
 
-    String partitionName = logRecordScanner.getPartitionName().get();
+    Map<String, Option<HoodieRecord<HoodieMetadataPayload>>> logRecords = new HashMap<>(keys.size());
 
-    Map<String, Option<HoodieRecord<HoodieMetadataPayload>>> logRecords = new HashMap<>();
-    if (isFullScanAllowedForPartition(partitionName)) {
```

Comment:
incase of MDT, only for FILES partition, we enable full scan but for other partitions we do on-demand calls. So, we can't really rely on forceFullScan instance var value in LogRecordReader. Not sure how this pans out inline 273 to 276 after this fix. bcoz, I dont' see isFullScanAllowedForPartition() being used after this patch. am I missing something

Reply:
Full-scan or not should be implementation detail of the Reader, and after this patch we don't need to know that to use it -- APIs now provide similar incremental semantic (only keys that weren't scanned before would be scanned)

Reply:
I meant something diff. but I checked the code, we are good.